SparkPost/2web2ui
# SA-1715 - (UI) Comparators - Update visible active filters for new comparators

[SA-1715](https://sparkpost.atlassian.net/browse/SA-1715)

### What Changed
- Updates review badger
- Adds new global `Comparison` text component - updates one-off component in `FiltersForm` with said component
- Adds new `ActiveFiltersV2` component to `ReportOptions`
- Adds new `removeFilterV2` that uses `reducerV2` in the report builder context hook
- Migrates report builder helper functions to a single file, adds JSDoc comments to clarify their purpose. I renamed a few of the helpers as well for clarity's sake.
- I tweaked a few Cypress tests while I was there to make them a little better - also added some missing `id` attributes to form elements in the saved reports modal. That was an a11y bug.

### How To Test
1. Verify filter tags render when filters are applied via the URL - append `query_filters=%255B%257B%2522AND%2522%3A%257B%2522campaigns%2522%3A%257B%2522like%2522%3A%255B%2522hello%2522%2C%2522world%2522%255D%257D%2C%2522templates%2522%3A%257B%2522eq%2522%3A%255B%2522greg-hackathon%2522%255D%2C%2522notEq%2522%3A%255B%2522gregs-test%2522%255D%257D%257D%257D%2C%257B%2522AND%2522%3A%257B%2522sending_ips%2522%3A%257B%2522like%2522%3A%255B%2522hello%2522%255D%2C%2522notLike%2522%3A%255B%2522hello-again%2522%255D%257D%257D%257D%255D` to the `/signals/analytics` URL
1. Use the `FiltersForm` to add and remove filters. Verify the tags update accordingly
1. Use the remove buttons within the tags and verify values are accurately removed
1. Verify the saved report modal renders the tags appropriately

### To Do
